packages/react-aria-components/src/ComboBox.tsx

```diff
@@ -41,11 +41,14 @@ function ComboBox<T extends object>(props: ComboBoxProps<T>, ref: ForwardedRef<H
   let state = useComboBoxState({
     defaultFilter: props.defaultFilter || contains,
     ...props,
-    items: propsFromListBox ? (props.items ?? propsFromListBox.items) : [],
+    // If props.items isn't provided, rely on collection filtering (aka listbox.items is provided or defaultItems provided to Combobox)
+    items: props.items,
     children: undefined,
     collection
   });
 
+  // Only expose a subset of state to renderProps function to avoid infinite render loop
+  let renderPropsState = useMemo(() => ({isOpen: state.isOpen, isFocused: state.isFocused}), [state.isOpen, state.isFocused]);
```

Comment:
For discussion: what subset of state should be exposed to renderProps for ComboBox? For other components with useRenderProps, what subset of state should be exposed?

TODO: update all components that support renderProps and update the docs accordingly when we decide on this

Reply:
my preference is just what is in the data-attributes which seems to be predominately the boolean values that we track, not what the user provides in props

Reply:
so comparing this to our docs, I wonder if there will be confusion between this state and say, the button's state.
https://react-spectrum.adobe.com/react-aria/ComboBox.html#button-1

For instance, if I wanted to style the focus for both the input and the button at the same time, do I use the button's state? or the overall combobox's? also, the props on that button also indicate the combobox button can have focus, but that's not true in a combobox, so does that reflect the combobox's state or its own? from looking at the code, I think it is only the button's state, not the combobox's

Reply:
yeah, thats a good point, I'm not entirely sure how we should handle this case. It almost feels like we should omit the isFocused+isFocusVisible parts of the button's state from the docs so that users don't try to use them. To be fair, the values returned from the button's renderProps are still accurate but I definitely could see some confusion arising from their usage.

I'm leaning towards making it clear in the docs that the combobox's/parent's renderProps should be preferred wherever possible.

Reply:
Talked with the rest of the the team, we are fine with the current docs state for the Button renderProps, will update if it proves to be confusing